digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/service/pointasset/TrafficLightService.scala

```diff
@@ -27,9 +27,9 @@ class TrafficLightService(val roadLinkService: RoadLinkService) extends PointAss
     }
   }
 
-  override def update(id: Long, updatedAsset: IncomingTrafficLight, geometry: Seq[Point], municipality: Int, username: String, linkSource: LinkGeomSource): Long = {
+  override def update(id: Long, updatedAsset: IncomingTrafficLight, roadLink: RoadLink, username: String): Long = {
     withDynTransaction {
-      updateWithoutTransaction(id, updatedAsset, geometry, municipality, username, linkSource, None, None)
+      updateWithoutTransaction(id, updatedAsset, roadLink.geometry, roadLink.municipalityCode, username, roadLink.linkSource, None, None)
```

Comment:
instead of , ...roadLink.geometry, roadLink.municipalityCode, username, roadLink.linkSource ...we could only as a parameter roadLink.